build.gradle.kts

```diff
@@ -38,27 +62,155 @@ version = File(project.rootDir.path + "/project.version").readLines().single()
 description = "A Java implementation of the Amazon Ion data notation."
 
 val isReleaseVersion: Boolean = !version.toString().endsWith("SNAPSHOT")
-val generatedJarInfoDir = "${buildDir}/generated/jar-info"
+val generatedResourcesDir = "$buildDir/generated/main/resources"
 lateinit var sourcesArtifact: PublishArtifact
 lateinit var javadocArtifact: PublishArtifact
 
 sourceSets {
     main {
         java.srcDir("src")
-        resources.srcDir(generatedJarInfoDir)
+        resources.srcDir(generatedResourcesDir)
     }
     test {
         java.srcDir("test")
     }
 }
 
+licenseReport {
+    // Because of the current gradle project structure, we must explicitly exclude ion-java-cli, even
+    // though ion-java does not depend on ion-java-cli. By default, the license report generator includes
+    // the current project (ion-java) and all its subprojects.
+    projects = arrayOf(project)
+    outputDir = "$buildDir/reports/licenses"
+    renderers = arrayOf(InventoryMarkdownReportRenderer(), TextReportRenderer())
+    // Dependencies use inconsistent titles for Apache-2.0, so we need to specify mappings
+    filters = arrayOf(
+        LicenseBundleNormalizer(
+            mapOf(
+                "The Apache License, Version 2.0" to "Apache-2.0",
+                "The Apache Software License, Version 2.0" to "Apache-2.0",
+            )
+        )
+    )
+}
+
 tasks {
     withType<JavaCompile> {
         options.encoding = "UTF-8"
         // Because we set the `release` option, you can no longer build ion-java using JDK 8. However, we continue to
         // emit JDK 8 compatible classes due to widespread use of this library with JDK 8.
         options.release.set(8)
     }
+    withType<KotlinCompile<KotlinJvmOptions>> {
+        kotlinOptions {
+            // Kotlin jvmTarget must match the JavaCompile release version
+            jvmTarget = "1.8"
+        }
+    }
+
+    jar {
+        archiveClassifier.set("original")
+    }
+
+    // Creates a super jar of ion-java and its dependencies where all dependencies are shaded (moved)
+    // to com.amazon.ion_.shaded.are_you_sure_you_want_to_use_this
+    shadowJar {
+        val newLocation = "com.amazon.ion.shaded_.are_you_sure_you_want_to_use_this"
```

Comment:
Should it be something more direct like `_do_not_use_this`?

Reply:
Sure. I can change it.